sharkfuser/tests/lit/test_conv_wgrad_asm_emitter_nhwc_krsc_grouped_strided.cpp

```diff
@@ -0,0 +1,168 @@
+// LINALG-CHECK:      %[[OUT_F:.+]] = linalg.fill ins(%[[CST]] : f32) outs(%[[E3_E]] : tensor<128x16x256x2x2xf32>) -> tensor<128x16x256x2x2xf32>
+// LINALG-CHECK:      %[[OUT:.+]] = linalg.conv_2d_ngchw_gfchw {dilations = dense<2> : vector<2xi64>, strides = dense<1> : vector<2xi64>} ins(%[[X_TE]], %[[DY_TE]]  : tensor<128x16x1x64x32xf32>, tensor<16x256x1x32x16xf32>) outs(%[[OUT_F]] : tensor<128x16x256x2x2xf32>) -> tensor<128x16x256x2x2xf32>
+// LINALG-CHECK:      %[[OUT_C:.+]] = tensor.collapse_shape %[[OUT]] {{\[\[0\], \[1, 2\], \[3\], \[4\]\]}} : tensor<128x16x256x2x2xf32> into tensor<128x4096x2x2xf32>
+// LINALG-CHECK:      %[[OUT_S1:.+]] = tensor.extract_slice %[[OUT_C:.+]][0, 0, 0, 0] [128, 4096, 1, 2] [1, 1, 1, 1] : tensor<128x4096x2x2xf32> to tensor<128x4096x1x2xf32>
+// LINALG-CHECK:      %[[OUT_S2:.+]] = tensor.extract_slice %[[OUT_S1:.+]][0, 0, 0, 0] [128, 4096, 1, 1] [1, 1, 1, 1] : tensor<128x4096x1x2xf32> to tensor<128x4096x1x1xf32>
+// LINALG-CHECK:      %[[OUT_E:.+]] = tensor.expand_shape %[[OUT_S2:.+]] {{\[\[0\], \[1, 2\], \[3\], \[4\]\]}} output_shape [128, 16, 256, 1, 1] : tensor<128x4096x1x1xf32> into tensor<128x16x256x1x1xf32>
+// LINALG-CHECK:      %[[OUT_5D:.+]] = tensor.empty() : tensor<16x256x128x1x1xf32>
+// LINALG-CHECK:      %[[OUT_T5D:.+]] = linalg.transpose ins(%[[OUT_E]] : tensor<128x16x256x1x1xf32>) outs(%[[OUT_5D]] : tensor<16x256x128x1x1xf32>) permutation = [1, 2, 0, 3, 4]
+// LINALG-CHECK:      %[[OUT_4D:.+]] = tensor.empty() : tensor<256x128x1x1xf32>
+// LINALG-CHECK:      %[[FILL:.+]] = linalg.fill ins(%[[CST]] : f32) outs(%[[OUT_4D]] : tensor<256x128x1x1xf32>) -> tensor<256x128x1x1xf32>
+// LINALG-CHECK:      %[[RED:.+]] = linalg.generic {indexing_maps = [affine_map<(d0, d1, d2, d3, d4) -> (d0, d1, d2, d3, d4)>, affine_map<(d0, d1, d2, d3, d4) -> (d1, d2, d3, d4)>], iterator_types = ["reduction", "parallel", "parallel", "parallel", "parallel"]} ins(%[[OUT_T5D]] : tensor<16x256x128x1x1xf32>) outs(%[[FILL]] : tensor<256x128x1x1xf32>) {
+// LINALG-CHECK:      ^bb0(%[[IN:.+]]: f32, %[[OUTV:.+]]: f32):
+// LINALG-CHECK:        %[[SUM:.+]] = arith.addf %[[IN]], %[[OUTV]] : f32
+// LINALG-CHECK:        linalg.yield %[[SUM]] : f32
+// LINALG-CHECK:      } -> tensor<256x128x1x1xf32>
+// LINALG-CHECK:      %[[CAST1:.+]] = tensor.cast %[[RED]] : tensor<256x128x1x1xf32> to tensor<?x?x?x?xf32>
+// LINALG-CHECK:      %[[CAST2:.+]] = tensor.cast %[[CAST1]] : tensor<?x?x?x?xf32> to tensor<256x16x1x1xf32>
+// LINALG-CHECK:      %[[OUT_BUF4:.+]] = tensor.empty() : tensor<256x1x1x16xf32>
+// LINALG-CHECK:      %[[OUT_T4:.+]] = linalg.transpose ins(%[[CAST2]] : tensor<256x16x1x1xf32>) outs(%[[OUT_BUF4]] : tensor<256x1x1x16xf32>) permutation = [0, 2, 3, 1]
+// LINALG-CHECK:      %{{.+}} = hal.tensor.alias wait(%{{.+}}) => %[[OUT_T4]] : tensor<256x1x1x16xf32> to %[[ARG0]] : !hal.buffer_view
+//
+// AMDGPU-STATS-CHECK: "dispatch-count": 2
+// CPU-STATS-CHECK: "dispatch-count": 2
```

Comment:
After lowering we get two dispatch regions: convolution and sum reduction by batch. As far as I understand, this is expected behavior for  backward not unit-stride convolutions with weight gradient calculations.

Reference in "DecomposeComplexOps" pass in torch-mlir: [link](https://github.com/iree-org/torch-mlir/blob/b8f742b2aa0fafa065d37198f574c4d23aabd886/lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp#L5824)
Logs (mlir, command line, lowered mlir): [link](https://gist.github.com/a-sidorova/054db1fb0adacaf94e8ef8f1760d3e52)

Reply:
@IanWood1 please verify if this is expected. If not, let's add a TODO with the issue link.